### PR TITLE
Fix false positives on rule `first_where`

### LINKED ISSUE #1930: first_where false positive when using with Realm Result container
```swift
// This triggers a violation:
realm?.objects(User.self).filter(NSPredicate(format: "email ==[c] %@", email)).first
```

### PR DESCRIPTION
**Pull request by Dschee**
Fixes #1930.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Performance/FirstWhereRule.swift [hunk 1]
```diff
@@ -16,7 +16,9 @@ public struct FirstWhereRule: CallPairRule, OptInRule, ConfigurationProviderRule
             "myList.first(where: { $0 % 2 == 0 })\n",
             "match(pattern: pattern).filter { $0.first == .identifier }\n",
             "(myList.filter { $0 == 1 }.suffix(2)).first\n",
-            "collection.filter(\"stringCol = '3'\").first"
+            "collection.filter(\"stringCol = '3'\").first",
+            "realm?.objects(User.self).filter(NSPredicate(format: \"email ==[c] %@\", email)).first",
+            "} else if let pause = timeTracker.pauses.filter(\"beginDate < %@\", beginDate).first {"
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Performance/FirstWhereRule.swift]: can you update this to be a valid Swift snippet? (not starting with `} else` and with a full body `{}`)

File: Source/SwiftLintFramework/Rules/Performance/FirstWhereRule.swift [hunk 2]
```diff
@@ -30,12 +32,18 @@ public struct FirstWhereRule: CallPairRule, OptInRule, ConfigurationProviderRule
...
+        return validate(
+            file: file,
+            pattern: "[\\}\\)]\\s*\\.first",
+            patternSyntaxKinds: [.identifier],
+            callNameSuffix: ".filter",
+            severity: configuration.severity
+        ) { dictionary in
+            if
+                !dictionary.substructure.isEmpty &&
+                dictionary.substructure.last!["key.kind"] as? String != "source.lang.swift.expr.argument" &&
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Performance/FirstWhereRule.swift]: 1. Can you remove the force unwraps?
2. We have extensions on `Dictionary` to get access to `key.kind` (`dictionary.kind`) and `key.name` (`dictionary.name`)
3. You can use the value from `SwiftExpressionKind`: `dictionary.substructure.last.kind.flatMap(SwiftExpressionKind.init) != .argument`

Comment (Dschee) [reply Source/SwiftLintFramework/Rules/Performance/FirstWhereRule.swift]: Done.

File: Source/SwiftLintFramework/Rules/Performance/FirstWhereRule.swift (hunk 1 above)
Comment (Dschee) [reply Source/SwiftLintFramework/Rules/Performance/FirstWhereRule.swift]: Oh, I didn't see this last time, just fixed it.

**Comment by Dschee** [comment]
@marcelofabri I think this can be merged, no? I fixed your comments quite a few days ago.

**Comment by Dschee** [comment]
No worries, thanks for approving! I'm not sure what the "perfect" procedure is here, can I just merge now given that you approved it and I myself don't see any reasons against merging? (Or to put it differently: Why don't you merge right after approving it?)

**Comment by marcelofabri** [comment]
I didn't merge to make sure you had the final word on whether it's ready to merge as I know you have push access and can merge it.

Sometimes a PR looks good to merge but the original author may prefer to not merge it (e.g. my own PR here https://github.com/realm/SwiftLint/pull/2633).

Feel free to merge a PR whenever it's approved.

**Comment by Dschee** [comment]
Thanks for the explanation. Merging. 🎉

**Pull request merged**